### ISSUE TITLE
Retrieving logits from model output

### ISSUE THREAD

**Issue by GuyAglionby** [stanfordnlp/dspy#184]
I'd like to be able to look at logits for each token for models where that's supported (e.g. HF models). Is this something you might support in the future? Thanks!

**Comment by okhat** [comment]
Thanks @GuyAglionby. I'm open to this but it's my bet that, increasingly, fewer infrastructure systems (e.g., VLLM etc) will give you this feature and it breaks compatibility of your program with many LMs.

Is it really necessary to see logits? Sometimes you can do even better by generating n=5 or n=10 outputs from the LM and using them for some kind of vote / count / comparison.

**Comment by GuyAglionby** [comment]
Thanks for getting back! I hadn't come across vLLM so thanks for the pointer. It [looks like](https://github.com/vllm-project/vllm/blob/1f24755bf802a2061bd46f3dd1191b7898f13f45/vllm/sampling_params.py#L62) they do support return of logits, but I haven't looked deeply so might have interpreted that wrong. To expand on why I ask, I'm interested in them as a proxy for confidence rather than as something that will interact with the program.

My main concern with running inference multiple times is computational cost, given that I have a lot of instances. I've not yet properly worked out if it would be prohibitively expensive but I suspect it will be, and I'm reluctant to go down that path given that there is this alternative.

I'm happy to send a (draft) PR when I get to this, if it would be welcomed :)
